CMakeLists.txt

```diff
@@ -184,68 +184,12 @@ set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> cqls <TARGET> <LINK_FLAGS> <OBJECTS>")
 # Global Includes
 #----------------------------------------------------------------
 # include paths to main abstract classes
- 
-# NOTE BeamIntegration and MatrixUtil need to be removed from element/forceBEamColumn
-
-include_directories(
-  ${OPS_SRC_DIR}
-  ${OPS_SRC_DIR}/matrix
-  ${OPS_SRC_DIR}/handler
-  ${OPS_SRC_DIR}/database
-  ${OPS_SRC_DIR}/element
-  ${OPS_SRC_DIR}/element/forceBeamColumn
-  ${OPS_SRC_DIR}/element/nonlinearBeamColumn/matrixutil
-  ${OPS_SRC_DIR}/coordTransformation
-  ${OPS_SRC_DIR}/damping
-  ${OPS_SRC_DIR}/tagged
-  ${OPS_SRC_DIR}/tagged/storage
-  ${OPS_SRC_DIR}/recorder
-  ${OPS_SRC_DIR}/renderer
-  ${OPS_SRC_DIR}/damage
-  ${OPS_SRC_DIR}/recorder/response
-  ${OPS_SRC_DIR}/material
-  ${OPS_SRC_DIR}/material/section
-  ${OPS_SRC_DIR}/material/uniaxial
-  ${OPS_SRC_DIR}/material/nD
-  ${OPS_SRC_DIR}/graph/graph
-  ${OPS_SRC_DIR}/graph/numberer
-  ${OPS_SRC_DIR}/graph/partitioner
-  ${OPS_SRC_DIR}/domain/component
-  ${OPS_SRC_DIR}/domain/domain
-  ${OPS_SRC_DIR}/domain/subdomain
-  ${OPS_SRC_DIR}/domain/load
-  ${OPS_SRC_DIR}/domain/loadBalancer
-  ${OPS_SRC_DIR}/domain/pattern
-  ${OPS_SRC_DIR}/domain/groundMotion
-  ${OPS_SRC_DIR}/domain/node
-  ${OPS_SRC_DIR}/domain/constraints
-  ${OPS_SRC_DIR}/domain/region
-  ${OPS_SRC_DIR}/analysis/algorithm
-  ${OPS_SRC_DIR}/analysis/dof_grp
-  ${OPS_SRC_DIR}/analysis/fe_ele
-  ${OPS_SRC_DIR}/analysis/algorithm/equiSolnAlgo
-  ${OPS_SRC_DIR}/analysis/algorithm/eigenAlgo
-  ${OPS_SRC_DIR}/analysis/algorithm/domainDecompAlgo
-  ${OPS_SRC_DIR}/analysis/analysis
-  ${OPS_SRC_DIR}/analysis/integrator
-  ${OPS_SRC_DIR}/analysis/handler
-  ${OPS_SRC_DIR}/analysis/numberer
-  ${OPS_SRC_DIR}/analysis/model
-  ${OPS_SRC_DIR}/convergenceTest
-  ${OPS_SRC_DIR}/modelbuilder
-  ${OPS_SRC_DIR}/system_of_eqn
-  ${OPS_SRC_DIR}/system_of_eqn/linearSOE
-  ${OPS_SRC_DIR}/system_of_eqn/eigenSOE
-  ${OPS_SRC_DIR}/actor/actor
-  ${OPS_SRC_DIR}/actor/channel
-  ${OPS_SRC_DIR}/actor/objectBroker
-  ${OPS_SRC_DIR}/actor/message
-)
-
-# TODO: Temporary fix; include all directories with .h files through glob
+# include all directories with CMakeLists.txt files through glob
+# Searching for *.h masks when CMakeLists.txt are not setup
+include_directories(${OPS_SRC_DIR})
 file(GLOB_RECURSE ops_include_files CONFIGURE_DEPENDS 
-    ${OPS_SRC_DIR}/*.h
-    ${OPS_BUNDLED_DIR}/*.h
+    ${OPS_SRC_DIR}/CMakeLists.txt
+    #${OPS_BUNDLED_DIR}/CMakeLists.txt
```

Comment:
This change does not make sense to me. Why are CMake files being globbed for includes? If the glob results are passed to `include_directories()`, shouldn't they be header files?

I did the globbing as a temporary patch and it should eventually be removed, as it is generally poor practice and doesn't let the build system make a proper dependency representation. Hard-coded includes are the best option for the core subdirectories. I've done this on my fork (https://github.com/claudioperez/OpenSeesRT/blob/master/CMakeLists.txt) which has significantly improved build quality, but I have not had a chance to merge it in.

However, for external libraries, it would be great to have user-configurable paths. I originally implemented a system for this, but things got messy and we fell back to a combination of `find_package` and/or Conan. I'd love to see some progress on this.